uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/leave-request-api_test.js

```diff
@@ -115,5 +147,259 @@ define([
         $httpBackend.flush();
       });
     });
+
+    describe('calculateBalanceChange()', function () {
+      var promise, requestData;
+
+      beforeEach(function () {
+        requestData = mockData.calculateBalanceChangeRequest();
+        spyOn(LeaveRequestAPI, 'calculateBalanceChange').and.callThrough();
+        promise = LeaveRequestAPI.calculateBalanceChange(requestData);
+      });
+
+      afterEach(function () {
+        $httpBackend.flush();
+      });
+
+      it('calls endpoint', function () {
+        promise.then(function (result) {
+          expect(LeaveRequestAPI.calculateBalanceChange).toHaveBeenCalled();
```

Comment:
i thought it was to test the endpoint call and then callthrough to the function (if the then succeeds then simply verify if it was due to the calculateBalanceChange call ), will have to discuss this